diff --git a/src/game/BattleGround/BattleGround.h b/src/game/BattleGround/BattleGround.h
index 00d55fd0c..a288203df 100644
--- a/src/game/BattleGround/BattleGround.h
+++ b/src/game/BattleGround/BattleGround.h
@@ -82,8 +82,8 @@ enum BattleGroundMarks
 
 enum BattleGroundMarksCount
 {
-    ITEM_WINNER_COUNT               = 3,
-    ITEM_LOSER_COUNT                = 1
+    ITEM_WINNER_COUNT               = 60,
+    ITEM_LOSER_COUNT                = 20
 };
 
 enum BattleGroundSpells
