diff --git a/playerbot/strategy/actions/CheckMountStateAction.cpp b/playerbot/strategy/actions/CheckMountStateAction.cpp
index d21441b4..21192e92 100644
--- a/playerbot/strategy/actions/CheckMountStateAction.cpp
+++ b/playerbot/strategy/actions/CheckMountStateAction.cpp
@@ -273,7 +273,7 @@ bool CheckMountStateAction::isUseful()
         return false;
 
 #ifndef MANGOSBOT_ZERO
-    if (bot->InArena())
+    if (bot->InArena() || (bot->InBattleGround() && sRandomPlayerbotMgr.GetValue(bot->GetGUID().GetCounter(), selfbot)))
         return false;
 #endif
 
