diff --git a/playerbot/PlayerbotFactory.cpp b/playerbot/PlayerbotFactory.cpp
index 2f03556f..86eeab33 100644
--- a/playerbot/PlayerbotFactory.cpp
+++ b/playerbot/PlayerbotFactory.cpp
@@ -3240,7 +3240,7 @@ void PlayerbotFactory::InitEquipment(bool incremental, bool syncWithMaster, bool
             else
             {
                 std::vector<uint32> ids;
-                for (uint32 q = quality; q < ITEM_QUALITY_ARTIFACT; ++q)
+                for (int q = quality; q < (int)ITEM_QUALITY_ARTIFACT; ++q)
                 {
                     // quality selected from command
                     if (setQuality && q != quality)
