diff --git a/playerbot/PlayerbotFactory.cpp b/playerbot/PlayerbotFactory.cpp
index 0000000..0000000 100644
--- a/playerbot/PlayerbotFactory.cpp
+++ b/playerbot/PlayerbotFactory.cpp
@@ -3243,7 +3243,7 @@ bool PlayerbotFactory::CanEquipItem(uint8 slot, uint8& specId, ItemPrototype con
             else
             {
                 std::vector<uint32> ids;
-                for (uint32 q = quality; q < ITEM_QUALITY_ARTIFACT; ++q)
+                for (int q = quality; q < (int)ITEM_QUALITY_ARTIFACT; ++q)
                 {
                     // quality selected from command
                     if (setQuality && q != quality)
